File: apps/admin/src/app/core/clients/client-summary.ts
export interface ClientSummary {
  readonly id: string;
  readonly name: string;
  readonly phone: string;
  readonly alternatePhone: string | null;
  readonly origin: string | null;
  readonly notes: string | null;
}

export interface CreateClientResult {
  readonly client: ClientSummary;
  readonly possibleDuplicate: boolean;
}

File: apps/api/src/main/java/br/com/agendaplatform/clients/application/ClientSummary.java
package br.com.agendaplatform.clients.application;

import br.com.agendaplatform.clients.domain.Client;
import java.util.UUID;

public record ClientSummary(
        UUID id,
        String name,
        String phone,
        String alternatePhone,
        String origin,
        String notes,
        boolean contactRestricted,
        String contactRestrictionReason) {

    static ClientSummary from(Client client) {
        return new ClientSummary(
                client.getId(),
                client.getName(),
                client.getPhone(),
                client.getAlternatePhone(),
                client.getOrigin(),
                client.getNotes(),
                client.isContactRestricted(),
                client.getContactRestrictionReason());
    }
}
